### ISSUE TITLE
zip4j 2.0: When execute ZipFile.getInputStream(FileHeader), do something and close the inputstream, the zipFile is still occupied.

### ISSUE THREAD

**Issue by dancencheung** [srikanth-lingala/zip4j#5]
zip4j 2.0: When execute ZipFile.getInputStream(FileHeader), do something and then close the inputstream, the zipFile is still occupied.

ZipFile zipFile = new ZipFile("filename.zip");
FileHeader fileHeader = zipFile.getFileHeader("filePathInZip");

if(null != fileHeader && !fileHeader.isDirectory())
{
    InputStream inputStream = zipFile.getInputStream(fileHeader);
    //TO DO SOMETHING
    inputStream.close();
    //after this, rename  the zipFile will failed, the file is occupied.
}

**Comment by srikanth-lingala** [comment]
I tried your code and was unable to reproduce this issue. The zip file gets successfully renamed

```
ZipFile zipFile = new ZipFile(generatedZipFile);
FileHeader fileHeader = zipFile.getFileHeader("sample.pdf");

if(null != fileHeader && !fileHeader.isDirectory())
{
  InputStream inputStream = zipFile.getInputStream(fileHeader);
  inputStream.read(new byte[50]);
  inputStream.close();

  System.out.println("Rename status: " + generatedZipFile.renameTo(new File(generatedZipFile.getParentFile() + InternalZipConstants.FILE_SEPARATOR + "sometthing.zip")));
  System.out.println("Done");
}
```

Maybe there is a handle existing for this file in some other part of your code?

**Comment by dancencheung** [comment]
I reconfirmed the problem. Below test code always prints out false for me.
Before execute the code, I can rename the zipFile successfully, and after execute the code, I can rename the zipFile successfully too. Obviously, this is a simple testcase, there is no other code exists.

`public class Test
{
	public static void main(String[] args)
	{
		try
		{
			File file = new File("D:\\v.zip");
			File renamedFile = new File("D:\\v_1.zip");
			ZipFile zipFile = new ZipFile(file);
			FileHeader fileHeader = zipFile.getFileHeader("v.xml");
			
			if(null != fileHeader && !fileHeader.isDirectory())
			{
				ZipInputStream zipInputStream = zipFile.getInputStream(fileHeader);
				zipInputStream.read(new byte[64]);
				zipInputStream.close();
				
				System.out.println("Rename status: " + file.renameTo(renamedFile));
				System.out.println("Done");
			}
		}
		catch(Exception e)
		{
			e.printStackTrace();
		}
	}
}`

I checked the method:
net.lingala.zip4j.io.inputstream.ZipInputStream.close()

The code is:
`  @Override
  public void close() throws IOException {
    if (decompressedInputStream != null) {
      decompressedInputStream.close();
    }
  }`

I made a change for it:
`  @Override
  public void close() throws IOException {
    if (decompressedInputStream != null) {
      decompressedInputStream.close();
    }
    
    if(null != inputStream)
    {
    	inputStream.close();
    }
  }`

Now the test code  always prints out true. Is there some problems for this method?

**Comment by srikanth-lingala** [comment]
I was still not able to reproduce this issue on a mac. But I think I got the problem.

I made a snapshot release `2.0.3-SNAPSHOT` with the fix. Can you please try and let me know if this fixes it? If yes, then I will make a final release of this version.

Note: You must have [snapshot repository configured in your maven settings][1] for maven to download this snapshot version.

[1]: https://stackoverflow.com/a/7717234/2736153

**Comment by dancencheung** [comment]
Thanks. Updated to 2.0.3-SNAPSHOT, the bug fixed.

**Comment by srikanth-lingala** [comment]
Fixed in v2.3.3
